salt/modules/cmdmod.py

```diff
@@ -245,6 +271,8 @@ def _run(cmd,
     '''
     Do the DRY thing and only call subprocess.Popen() once
     '''
+    if 'pillar' in kwargs and not pillar_override:
+        pillar_override = kwargs['pillar']
```

Comment:
Yes, good point, I would think though that this would be easier since the logic was spread out to a bunch of other calls before